Bottom: e19c883629f0add91e22d5dad478ebda71560b35
Top:    78017a4eeb8169b9d957905e25b0804386cb4345
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-24 11:59:23 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index c77ee2c..e427f1a 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -309,6 +309,16 @@ original_poly (PolygonType * p)
       if (n == p->PointN - 1 ||
           (hole < p->HoleIndexN && n == p->HoleIndex[hole] - 1))
         {
+          if (contour == NULL)
+            {
+              printf ("How did that escape - did the loop iterate zero times??\n");
+              POLYGONPOINT_LOOP (p);
+                {
+                  printf ("Hello\n");
+                }
+              END_LOOP;
+              return NULL;
+            }
           poly_PreContour (contour, TRUE);
 
           /* make sure it is a positive contour (outer) or negative (hole) */
